src/lib.rs

```diff
@@ -260,78 +253,66 @@ fn compute_nearest_normal(
     >(zi);
     let mut r = (zi - BIG_DIVISOR as u64 * significand) as u32;
 
-    'small_divisor_case_label: loop {
-        if r > deltai {
-            break 'small_divisor_case_label;
-        } else if r < deltai {
+    loop {
+        if r < deltai {
             // Exclude the right endpoint if necessary.
-            if r == 0
-                && !has_even_significand_bits
-                && is_product_integer_fc_pm_half(two_fr, exponent, minus_k)
-            {
+            if r == 0 && !z_result.is_integer && !include_right_endpoint {
```

Comment:
Hello,
According to the original code
```c++
                            if ((r | remainder_type_(!z_result.is_integer) |
                                 remainder_type_(interval_type.include_right_endpoint())) == 0) {
```
Should this line to be the following?
```suggestion
            if r == 0 && z_result.is_integer && !include_right_endpoint {
```

Reply:
Thank you!